readthedocs/oauth/services/base.py

```diff
@@ -131,6 +138,26 @@ def paginate(self, url, **kwargs):
         """
         try:
             resp = self.get_session().get(url, data=kwargs)
+
+            # TODO: this check of the status_code would be better in the
+            # ``create_session`` method since it could be used from outside, but
+            # I didn't find a generic way to make a test request to each
+            # provider.
+            if resp.status_code == 401:
+                # Bad credentials: the token we have in our database is not
+                # valid. Probably the user has revoked the access to our App. He
+                # needs to reconnect his account
+                message = 'Our access to your {} account was revoked. You ' \
```

Comment:
Consdider to use a better general message here and also maybe catch the 404 --in a another PR with more testing, probably.